# 690IndividualProject_Part-B
  Name: Ramteja Ponnaagnti


In [88]:
import pandas as pd
import plotly.graph_objects as go
import plotly.express as px
import plotly.io as pio

In [89]:
pio.templates

Templates configuration
-----------------------
    Default template: 'plotly_dark'
    Available templates:
        ['ggplot2', 'seaborn', 'simple_white', 'plotly',
         'plotly_white', 'plotly_dark', 'presentation', 'xgridoff',
         'ygridoff', 'gridon', 'none']

In [90]:
pio.templates.default = "plotly_dark"

In [91]:
import plotly

print(plotly.__version__)

5.9.0


In [92]:
!pip install plotly --upgrade

In [93]:
print(plotly.__version__)

5.9.0


In [94]:
df = pd.read_excel("Wdi_Data.xlsx") #importing data
df.head(10)

Year         value          indicator Country Code Country Name  \
0  2015   3555.505260  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
1  2016   3848.957664  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
2  2017   4160.703325  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
3  2018   4547.479572  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
4  2019   4954.761400  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
5  2020   5082.680750  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
6  2015  12978.768911  NY.GDP.PCAP.PP.CD          CHN        China   
7  2016  13572.619989  NY.GDP.PCAP.PP.CD          CHN        China   
8  2017  14344.421239  NY.GDP.PCAP.PP.CD          CHN        China   
9  2018  15608.967069  NY.GDP.PCAP.PP.CD          CHN        China   

                Region         Income Group Lending Type  
0           South Asia  Lower middle income          IDA  
1           South Asia  Lower middle income          IDA  
2           South Asia  Lower middle income          IDA  
3           South Asia  Lower middle income          IDA  
4           South Asia  Lower middle income          IDA  
5           South Asia  Lower middle income          IDA  
6  East Asia & Pacific  Upper middle income         IBRD  
7  East Asia & Pacific  Upper middle income         IBRD  
8  East Asia & Pacific  Upper middle income         IBRD  
9  East Asia & Pacific  Upper middle income         IBRD

# Topics Included:
- Economy and Growth
- Health
- Environment

# Indicators Included:
- GDP per capita, PPP (current international $): This indicator falls under the topic of economy and growth. Gross Domestic Product, or GDP, is defined as GDP divided by the midyear population. It describes the economic production of the nation.
- Population, total: This indicator is a part of the Health subject, which specifies the total population by year for each nation.
- Cause of death, by communicable diseases and maternal, prenatal and nutrition conditions (% of total): This indicator, which is a part of the Health topic, shows how many people have died overall as a result of communicable diseases and malnutrition.
- Mortality rate attributed to unsafe water, unsafe sanitation, and lack of hygiene (per 100,000 population): This indicator is a part of the Environment subject, which details the global disease and mortality rates.

# Countries Included:
- India
- United States
- China
- United Kingdom
- Spain
- Bangladesh

# Time Period Included:
By the guidance of Prof. Wang, the period of over 5 years starting from the year 2015 to the ending of the year 2020 has been selected.

# Intoduction:
The technology is expanding quickly in the modern world. As a result of that technology, our environment is likewise undergoing significant changes. These modifications may be the result of modernization or a shift toward focusing more on development than the environment.

This document provides a detailed description of how these developments have affected both the people's health and the state of the economy.

In [95]:
df_GDP = df.query('indicator == "NY.GDP.PCAP.PP.CD"') #query for GDP per capita data
df_GDP.head(10)

Year         value          indicator Country Code Country Name  \
0  2015   3555.505260  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
1  2016   3848.957664  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
2  2017   4160.703325  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
3  2018   4547.479572  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
4  2019   4954.761400  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
5  2020   5082.680750  NY.GDP.PCAP.PP.CD          BGD   Bangladesh   
6  2015  12978.768911  NY.GDP.PCAP.PP.CD          CHN        China   
7  2016  13572.619989  NY.GDP.PCAP.PP.CD          CHN        China   
8  2017  14344.421239  NY.GDP.PCAP.PP.CD          CHN        China   
9  2018  15608.967069  NY.GDP.PCAP.PP.CD          CHN        China   

                Region         Income Group Lending Type  
0           South Asia  Lower middle income          IDA  
1           South Asia  Lower middle income          IDA  
2           South Asia  Lower middle income          IDA  
3           South Asia  Lower middle income          IDA  
4           South Asia  Lower middle income          IDA  
5           South Asia  Lower middle income          IDA  
6  East Asia & Pacific  Upper middle income         IBRD  
7  East Asia & Pacific  Upper middle income         IBRD  
8  East Asia & Pacific  Upper middle income         IBRD  
9  East Asia & Pacific  Upper middle income         IBRD

In [96]:
df_pivot1 = df_GDP.pivot(index =['Year','Country Code','Country Name','Region','Income Group','Lending Type'],columns = 'indicator', values ='value').reset_index()
df_pivot1.reset_index().head()

indicator  index  Year Country Code    Country Name                 Region  \
0              0  2015          BGD      Bangladesh             South Asia   
1              1  2015          CHN           China    East Asia & Pacific   
2              2  2015          ESP           Spain  Europe & Central Asia   
3              3  2015          GBR  United Kingdom  Europe & Central Asia   
4              4  2015          IND           India             South Asia   

indicator         Income Group    Lending Type  NY.GDP.PCAP.PP.CD  
0          Lower middle income             IDA        3555.505260  
1          Upper middle income            IBRD       12978.768911  
2                  High income  Not classified       34903.127478  
3                  High income  Not classified       42567.591797  
4          Lower middle income            IBRD        5464.859178

# How does GDP per capita and Mortality rate due to unsafe conditions relate?
## GDP per capita by year in different countries:

In [97]:
fig = px.line(
    df_pivot1, 
    x="Year", 
    y="NY.GDP.PCAP.PP.CD",
    labels={"NY.GDP.PCAP.PP.CD":"GDP per capita, PPP (current international $)"},
    hover_name='Country Name',
    color='Country Name'
    )
fig.show() #line graph of GDP per capita by year

From the above time series graph the observations are as follows:
- When compared to other nations, Bangladesh's GDP per capita has been remarkably stable between the years 2015 and 2020.
- For both China and India, the pattern has been nearly the same. Over the years, they stayed the same in constancy.
- But in the cases of the USA, Spain, and the UK, the situation was different.
- During the first few years, we observe an increase in their GDP, but during the following years, the GDP of each country decelerated. This might be the result of a significant shift in the economy or a reduction in the sources that can drive it.

In [55]:
df_2016 = df.query("Year == 2016")
df_2016.sample(10)

Year         value          indicator Country Code    Country Name  \
7   2016  1.357262e+04  NY.GDP.PCAP.PP.CD          CHN           China   
85  2016  3.230718e+08        SP.POP.TOTL          USA   United States   
61  2016  1.378665e+09        SP.POP.TOTL          CHN           China   
25  2016  4.415000e+04  NY.GDP.PCAP.PP.CD          GBR  United Kingdom   
67  2016  1.324517e+09        SP.POP.TOTL          IND           India   
50  2016  1.860000e+01     SH.STA.WASH.P5          IND           India   
48  2016  1.190000e+01     SH.STA.WASH.P5          BGD      Bangladesh   
13  2016  5.839865e+03  NY.GDP.PCAP.PP.CD          IND           India   
79  2016  6.561159e+07        SP.POP.TOTL          GBR  United Kingdom   
1   2016  3.848958e+03  NY.GDP.PCAP.PP.CD          BGD      Bangladesh   

                   Region         Income Group    Lending Type  
7     East Asia & Pacific  Upper middle income            IBRD  
85          North America          High income  Not classified  
61    East Asia & Pacific  Upper middle income            IBRD  
25  Europe & Central Asia          High income  Not classified  
67             South Asia  Lower middle income            IBRD  
50             South Asia  Lower middle income            IBRD  
48             South Asia  Lower middle income             IDA  
13             South Asia  Lower middle income            IBRD  
79  Europe & Central Asia          High income  Not classified  
1              South Asia  Lower middle income             IDA

In [56]:
df_2016 = df_2016[df_2016["indicator"].isin(['SH.STA.WASH.P5'])]
df_2016.head()

Year  value       indicator Country Code    Country Name  \
48  2016   11.9  SH.STA.WASH.P5          BGD      Bangladesh   
49  2016    0.6  SH.STA.WASH.P5          CHN           China   
50  2016   18.6  SH.STA.WASH.P5          IND           India   
51  2016    0.2  SH.STA.WASH.P5          ESP           Spain   
52  2016    0.2  SH.STA.WASH.P5          GBR  United Kingdom   

                   Region         Income Group    Lending Type  
48             South Asia  Lower middle income             IDA  
49    East Asia & Pacific  Upper middle income            IBRD  
50             South Asia  Lower middle income            IBRD  
51  Europe & Central Asia          High income  Not classified  
52  Europe & Central Asia          High income  Not classified

In [58]:
df_pivot = df_2016.pivot(index =['Country Code','Country Name','Region','Income Group','Lending Type'],columns = 'indicator', values ='value').reset_index()
df_pivot.reset_index().head()

indicator  index Country Code    Country Name                 Region  \
0              0          BGD      Bangladesh             South Asia   
1              1          CHN           China    East Asia & Pacific   
2              2          ESP           Spain  Europe & Central Asia   
3              3          GBR  United Kingdom  Europe & Central Asia   
4              4          IND           India             South Asia   

indicator         Income Group    Lending Type  SH.STA.WASH.P5  
0          Lower middle income             IDA            11.9  
1          Upper middle income            IBRD             0.6  
2                  High income  Not classified             0.2  
3                  High income  Not classified             0.2  
4          Lower middle income            IBRD            18.6

# Mortality rate due to unsafe water, unsafe sanitation, and lack of hygiene in different countries:
Simply put, mortality rate is a measurement of how frequently people die in a given population over a certain time period. It is a mathematical representation that is used to calculate how many people get sick or die each year.

In [59]:
fig = px.bar(
    df_pivot,
    x="Country Name",
    y="SH.STA.WASH.P5",
    labels={"SH.STA.WASH.P5":"2016Mortality rate attributed to unsafe water, unsafe sanitation, and lack of hygiene (per 100,000 population)"},
    color="Country Name")
fig.update_xaxes(categoryorder="total descending")
fig.show() #bar graph of mortality rate

The following information can be depicted from the above bar graph:

- When compared to other nations, India had the highest mortality rate in 2016 with 18.6.
- China, the United States, and Spain had lower mortality rates with 0.2 each. The exploitations that people are doing in the name of progress are to blame for these illnesses or fatalities. For instance, when a new industrial is built inside a city, the residents who live nearby suffer greatly.

Additionally, waste from these groups needs to be controlled and directed to predetermined locations rather than open water. We can see in the graph above that there have been a few fatalities related to contaminated water, primarily as a result of industrial contamination. This is just one situation; there are countless others that occur frequently all around us.

In [60]:
df_Deaths = df.query('indicator == "SH.DTH.COMM.ZS"')
df_Deaths.head(10)

Year      value       indicator Country Code    Country Name  \
36  2015  28.807663  SH.DTH.COMM.ZS          BGD      Bangladesh   
37  2019  22.612837  SH.DTH.COMM.ZS          BGD      Bangladesh   
38  2015   4.132797  SH.DTH.COMM.ZS          CHN           China   
39  2019   3.549711  SH.DTH.COMM.ZS          CHN           China   
40  2015  28.260253  SH.DTH.COMM.ZS          IND           India   
41  2019  24.166184  SH.DTH.COMM.ZS          IND           India   
42  2015   5.324214  SH.DTH.COMM.ZS          ESP           Spain   
43  2019   5.610445  SH.DTH.COMM.ZS          ESP           Spain   
44  2015   8.066941  SH.DTH.COMM.ZS          GBR  United Kingdom   
45  2019   8.156775  SH.DTH.COMM.ZS          GBR  United Kingdom   

                   Region         Income Group    Lending Type  
36             South Asia  Lower middle income             IDA  
37             South Asia  Lower middle income             IDA  
38    East Asia & Pacific  Upper middle income            IBRD  
39    East Asia & Pacific  Upper middle income            IBRD  
40             South Asia  Lower middle income            IBRD  
41             South Asia  Lower middle income            IBRD  
42  Europe & Central Asia          High income  Not classified  
43  Europe & Central Asia          High income  Not classified  
44  Europe & Central Asia          High income  Not classified  
45  Europe & Central Asia          High income  Not classified

In [61]:
df_pivot2 = df_Deaths.pivot(index =['Year','Country Code','Country Name','Region','Income Group','Lending Type'],columns = 'indicator', values ='value').reset_index()
df_pivot2.reset_index().head()

indicator  index  Year Country Code    Country Name                 Region  \
0              0  2015          BGD      Bangladesh             South Asia   
1              1  2015          CHN           China    East Asia & Pacific   
2              2  2015          ESP           Spain  Europe & Central Asia   
3              3  2015          GBR  United Kingdom  Europe & Central Asia   
4              4  2015          IND           India             South Asia   

indicator         Income Group    Lending Type  SH.DTH.COMM.ZS  
0          Lower middle income             IDA       28.807663  
1          Upper middle income            IBRD        4.132797  
2                  High income  Not classified        5.324214  
3                  High income  Not classified        8.066941  
4          Lower middle income            IBRD       28.260253

# How does mortality rate relate with total population and number of deaths that are caused by the communicable diseases?

In [62]:
fig = px.line(
    df_pivot2.dropna(),
    x="Year", 
    y="SH.DTH.COMM.ZS",
    color='Country Name',
    hover_name = "Country Name",
    labels={"SH.DTH.COMM.ZS":"Cause of death, by communicable diseases and maternal, prenatal and nutrition conditions (% of total)"})
fig.show() 

In [63]:
df_2015 = df.query("Year == 2015")
df_2015.sample(10)

Year         value          indicator Country Code    Country Name  \
24  2015  4.256759e+04  NY.GDP.PCAP.PP.CD          GBR  United Kingdom   
12  2015  5.464859e+03  NY.GDP.PCAP.PP.CD          IND           India   
60  2015  1.371220e+09        SP.POP.TOTL          CHN           China   
72  2015  4.644483e+07        SP.POP.TOTL          ESP           Spain   
40  2015  2.826025e+01     SH.DTH.COMM.ZS          IND           India   
36  2015  2.880766e+01     SH.DTH.COMM.ZS          BGD      Bangladesh   
66  2015  1.310152e+09        SP.POP.TOTL          IND           India   
30  2015  5.686337e+04  NY.GDP.PCAP.PP.CD          USA   United States   
38  2015  4.132797e+00     SH.DTH.COMM.ZS          CHN           China   
84  2015  3.207390e+08        SP.POP.TOTL          USA   United States   

                   Region         Income Group    Lending Type  
24  Europe & Central Asia          High income  Not classified  
12             South Asia  Lower middle income            IBRD  
60    East Asia & Pacific  Upper middle income            IBRD  
72  Europe & Central Asia          High income  Not classified  
40             South Asia  Lower middle income            IBRD  
36             South Asia  Lower middle income             IDA  
66             South Asia  Lower middle income            IBRD  
30          North America          High income  Not classified  
38    East Asia & Pacific  Upper middle income            IBRD  
84          North America          High income  Not classified

In [66]:
df_pivot = df_2015.pivot(index =['Country Code','Country Name','Region','Income Group','Lending Type'],columns = 'indicator', values ='value').reset_index()
df_pivot.reset_index().head()

indicator  index Country Code    Country Name                 Region  \
0              0          BGD      Bangladesh             South Asia   
1              1          CHN           China    East Asia & Pacific   
2              2          ESP           Spain  Europe & Central Asia   
3              3          GBR  United Kingdom  Europe & Central Asia   
4              4          IND           India             South Asia   

indicator         Income Group    Lending Type  NY.GDP.PCAP.PP.CD  \
0          Lower middle income             IDA        3555.505260   
1          Upper middle income            IBRD       12978.768911   
2                  High income  Not classified       34903.127478   
3                  High income  Not classified       42567.591797   
4          Lower middle income            IBRD        5464.859178   

indicator  SH.DTH.COMM.ZS   SP.POP.TOTL  
0               28.807663  1.562563e+08  
1                4.132797  1.371220e+09  
2                5.324214  4.644483e+07  
3                8.066941  6.511622e+07  
4               28.260253  1.310152e+09

In [67]:
fig = px.pie(df_pivot, values='SP.POP.TOTL', names='Country Name', title='Total Population',hover_name='Country Name')
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

The above observations of mortality rate and total population are as follows:

- China has the largest population, but it has a very low number of mortality from communicable diseases, maternal, prenatal, and nutrition-related problems. This has demonstrated that even though the nation has the greatest population, it is still possible to maintain good health by eating the right foods.
- The stats for India, though, are pretty intriguing. India has the second-largest population and, at first, the biggest number of fatalities in 2015. Fortunately, these numbers have decreased somewhat each year since 2010, and they have continued to decline as of 2015.
- It all depends on the GDP of the nation because adopting a healthy diet can cut down on these fatalities. There are fewer communicable mortality in rich countries and more cases in less developed countries because of the need to allocate some funds for this.
- Although Bangladesh has a larger population than some other nations, it has more deaths there, and its GDP is virtually significantly lower.

In [68]:
df = pd.read_excel("Wdx_Data.xlsx")
df

Year  SH.DTH.IMRT.FE  SH.DTH.IMRT.MA  SP.POP.TOTL Country Code  \
0   1995          128977          156140    115169930          BGD   
1   1996          122763          148938    117649932          BGD   
2   1997          117010          141484    120160564          BGD   
3   1998          111205          134253    122682815          BGD   
4   1999          105593          127226    125189651          BGD   
..   ...             ...             ...          ...          ...   
61  2001           12250           15654    284968955          USA   
62  2002           12200           15645    287625193          USA   
63  2003           12137           15702    290107933          USA   
64  2004           12167           15669    292805298          USA   
65  2005           12159           15611    295516599          USA   

     Country Name         Region         Income Group    Lending Type  
0      Bangladesh     South Asia  Lower middle income             IDA  
1      Bangladesh     South Asia  Lower middle income             IDA  
2      Bangladesh     South Asia  Lower middle income             IDA  
3      Bangladesh     South Asia  Lower middle income             IDA  
4      Bangladesh     South Asia  Lower middle income             IDA  
..            ...            ...                  ...             ...  
61  United States  North America          High income  Not classified  
62  United States  North America          High income  Not classified  
63  United States  North America          High income  Not classified  
64  United States  North America          High income  Not classified  
65  United States  North America          High income  Not classified  

[66 rows x 9 columns]

In [82]:
df_1995 = df[df["Year"]==1995]
df_1995.shape

(6, 9)

In [83]:
df_2005 = df[df["Year"]==2005]
df_2005.shape

(6, 9)

Comparison of Female vs Male Death Counts in 1995

In [85]:
fig = px.scatter(data_frame=df_1995, 
             x="SH.DTH.IMRT.FE", y="SH.DTH.IMRT.MA", color="Country Name",  
             title="Comparison of Female vs Male Death Counts in 1995"
             )
fig.show()

Comparison of Female vs Male Death Counts in 2005

In [87]:
fig = px.scatter(data_frame=df_2005, 
             x="SH.DTH.IMRT.FE", y="SH.DTH.IMRT.MA", color="Country Name",  
             title="Comparison of Female vs Male Death Counts in 2005"
             )
fig.show()

In the above 2D scatterplots, we can observe the comparison of the female and male infant death counts starting in 1995 and the overall death counts at the end of the decade. i.e 2005.


# Conclusion:



These conclusions are drawn from data that has been compared across a small number of nations. The country's GDP determines whether it is in a developed, underdeveloped, or less developed stage. Nations must work to raise their GDP in order to experience progress in all spheres. We can even take into account more nations to try to determine the extent to which the GDP component is important on a global scale.